packages/core-database/lib/interface.js

```diff
@@ -303,7 +303,7 @@ module.exports = class ConnectionInterface {
           this.blocksInCurrentRound = []
           // TODO: find a betxter place to call this as this
           // currently blocks execution but needs to be updated every round
-          // this.walletManager.updateDelegates()
+          this.walletManager.updateDelegates()
```

Comment:
Yeah but how big is the impact at this point, something might have changed since that method was added quite long ago.

Reply:
Yeh I'm not sure. I can imagine it's intensive on mainnet though with thousands of wallets, and it feels to me like it'd make more sense to update the votebalance, etc as and when the txs come in instead of at each round. It wouldn't affect forging order since that's updated at the end of each round/start of next

Reply:
I initially updated the vote balance & delegate ranks in the `applyBlock` and `revertBlock` method but don't remember why I moved it out of those as it was implemented right before public release.

Reply:
I guess it's a little complex, because every tx someone sends, needs to update the delegate votebalance because they have just used some balance for the fees. If there was no performance impact I'd leave as is 🤔

Reply:
The main problem for performance is the sync because the delegate stats would need to be update a million times as opposed to a node that is just running after sync where the method has no impact on performance at all.

Reply:
Do you still need to run `this.walletManager.updateDelegates()` if the node is out of sync?